### DISCUSSION TITLE
Upgrade to 2.3.100 errors with remote nodes upgrading salt-minion

### DISCUSSION THREAD

**Discussion by Susfubar** [Security-Onion-Solutions/securityonion#7060] [Unsupported Versions] [answered]
Upgraded distributed setup to 2.3.100, only 1 forward node successfully upgraded. After letting sit and rebooting 2 other nodes they no longer come up. so-status shows services never come up and sitting in WAIT_START
. 
On the ones that didnt upgrade properly it shows this with the command. "sudo salt-call state.apply ssl -linfo"
![image](https://user-images.githubusercontent.com/89588946/152059194-9970327a-e276-4491-b89e-85767feaf0db.png)

Checking versions of salt on all nodes shows only manager and the 1 forward node on 3004. 
![image](https://user-images.githubusercontent.com/89588946/152059331-9bf88430-95a0-42ad-9586-50f05d2c4596.png)

**Comment by m0duspwnens** [comment, marked as answer]
For the ones that the salt-minion hasn't upgraded, they will not be allowed to run the ssl state until the salt-minion package has upgraded to 3004. For the nodes that Salt has not yet upgraded on, they won't try to upgrade until they run their scheduled highstate which is every 15 minutes by default. You can force the upgrade by running the following from the manager `salt <target_node> state.highstate` i.e.`salt trd-ids-es-s1_searchnode state.highstate`.

**Reply by Susfubar** [reply]
I have run highstate a few times, I sometimes have gotten some "minion did not return" errors but otherwise successful. Same with doing a `salt-call state.highstate` from the FWD node itself. Still no changes.

**Reply by m0duspwnens** [reply]
From the forward node, if you run `salt-call state.highstate -linfo` is there anything that indicates failure for why the minion cannot be upgraded? Also, from the forward node, what is returned if you run `salt-call pillar.get global:managerupdate` and which files are in `/etc/yum.repos.d/`?

**Reply by Susfubar** [reply]
running ubuntu 18.04
Believe this is what you are wanting for first command.

![image](https://user-images.githubusercontent.com/89588946/152065783-f468e6fe-05a2-4eb8-8071-9053318f4c96.png)

Second command

> local:
> 1

No such directory as yum.repos.d.

**Reply by Susfubar** [reply]
I do see this in a  `apt update`

> W: Failed to fetch https://repo.saltproject.io/py3/ubuntu/18.04/amd64/3004/dists/bionic/InRelease  Reading from proxy failed - read (115: Operation now in progress) [IP: 10.x.y.z 3142]
W: Failed to fetch https://repo.saltstack.com/py3/ubuntu/18.04/amd64/archive/3003/dists/bionic/InRelease  Reading from proxy failed - read (115: Operation now in progress) [IP: 10.x.y.z 3142]
W: Some index files failed to download. They have been ignored, or old ones used instead.

**Reply by m0duspwnens** [reply]
After that is run, is the salt-minion still not upgraded? If that is the case, can you run `apt-get update` on the forward node then run `salt-call state.highstate -l info` again?

**Reply by Susfubar** [reply]
Correct, still on 3003 even after running it and you can see my response above that apt update does have a "Reading from proxy failed"

**Reply by m0duspwnens** [reply]
Is that IP of the proxy your manager node or is is a proxy in your environment?

**Reply by Susfubar** [reply]
That IP is the manager node.

**Reply by m0duspwnens** [reply]
For the forward node that has successfully updated Salt, is it running Ubuntu also, or is it running CentOS? What OS are the other nodes running in your environment?

On the manager, what is the output of `so-status`? Also, on the manager, do you see your nodes IPs in the output of `iptables -nL | grep 3142`

**Reply by Susfubar** [reply]
All ubuntu Bionic Beaver, manager so-status is all green. Yes the IPs are in that table.

 One thing I do notice is the GRID screen is not showing the nodes that have issue connecting.

**Reply by Susfubar** [reply]
Tested on one that was still on the GRID list but not updated and has same symptoms as my forward node that isnt working.  Rebooted it and still has issue as well. 

apt update show proxy read fail and this is the so-status on forward and elastic nodes. 
![image](https://user-images.githubusercontent.com/89588946/152072240-7d0e1140-c35b-412d-b5e0-8e80bfcec441.png)

**Reply by m0duspwnens** [reply]
I will have to try to replicate this in the lab in the morning. In the meantime, if the nodes have direct access to the internet, you try to modify `/etc/apt/apt.conf.d/00Proxy` on the nodes that aren't upgrading and comment out `Acquire::http::Proxy` and `Acquire::https::Proxy` then run `apt-get update`. If that is successful, try to run `salt-call state.highstate` again and verify if the salt-minion upgraded.

If the above doesn't work, can you provide the output of `ls /etc/apt/sources.list.d/`? If there are files with the name `salt` in them, can you provide the output from running `cat <filename>`.

Lastly, are there any differences in these files on the forward node that did upgrade and the nodes that aren't upgrading Salt?

**Reply by m0duspwnens** [reply]
Which version of Security Onion were these originally installed on?

**Reply by Susfubar** [reply]
This did fix my issue! Removing the proxy from apt then doing a highstate. I see so-status returning green now and it returning to manager. Going to do this with the rest. I believe the original version we had when we built was 2.3.30. Unsure though.

**Reply by m0duspwnens** [reply]
That is good news.

**Reply by Susfubar** [reply]
Appreciate the help, odd issue. Not sure if this issue was just my environment or not.
